File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/ICesiumIntervalListWriter.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 An interface to an object that writes a list of intervals for a CZML
 property.
 

 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public interface ICesiumIntervalListWriter extends ICesiumElementWriter {
    /**
    *  
    Opens a writer to write information about a single interval.
    
    

    * @return The interval writer.
    */
    @Nonnull
    ICesiumPropertyWriter openInterval();

    /**
    *  
    Opens a writer to write information about a single interval.
    
    
    
    

    * @param start The start of the interval of time covered by this interval element.
    * @param stop The end of the interval of time covered by this interval element.
    * @return The interval writer.
    */
    @Nonnull
    ICesiumPropertyWriter openInterval(@Nonnull JulianDate start, @Nonnull JulianDate stop);
}

File: DotNet/CesiumLanguageWriter/Advanced/ICesiumIntervalListWriter.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// An interface to an object that writes a list of intervals for a CZML
    /// property.
    /// </summary>
    public interface ICesiumIntervalListWriter : ICesiumElementWriter
    {
        /// <summary>
        /// Opens a writer to write information about a single interval.
        /// </summary>
        /// <returns>The interval writer.</returns>
        [NotNull]
        ICesiumPropertyWriter OpenInterval();

        /// <summary>
        /// Opens a writer to write information about a single interval.
        /// </summary>
        /// <param name="start">The start of the interval of time covered by this interval element.</param>
        /// <param name="stop">The end of the interval of time covered by this interval element.</param>
        /// <returns>The interval writer.</returns>
        [NotNull]
        ICesiumPropertyWriter OpenInterval(JulianDate start, JulianDate stop);
    }
}